spec/spec_helper.rb

```diff
@@ -24,6 +24,13 @@
     end
   end
 
+  # Sidekiq-Pro related specs require license set in Bundler
+  unless Bundler.settings["gems.contribsys.com"]&.include?(":")
+    config.define_derived_metadata(sidekiq_pro: true) do |metadata|
+      metadata[:skip] = "Sidekiq::Pro license not found"
+    end
+  end
```

Comment:
These changes worked, except the Sidekiq Pro specs were also running for Sidekiq v6.5, which isn't supported. This should prevent that

```suggestion
  # Sidekiq-Pro related specs require license set in Bundler
  unless Bundler.settings["gems.contribsys.com"]&.include?(":") &&
         Gem::Version.new(Sidekiq::VERSION) >= Gem::Version.new("7.0.0")
    config.define_derived_metadata(sidekiq_pro: true) do |metadata|
      metadata[:skip] = "Sidekiq::Pro license not found or not supported"
    end
  end
```